src/actions/guides/database/testing.cr

```diff
@@ -19,18 +31,137 @@ class Guides::Database::Testing < GuideAction
+    that are nilable by default.
+
+    ### Sequences
+
+    Your model may have a unique constraint on a field like a `Post` title, for example. In this case,
+    you can use the `sequence` method to auto-increment a number on to your default value ensuring it will
+    be unique.
+
+    ```crystal
+    class PostBox < Avram::Box
+      def initialize
+        title sequence("My new blog post")
+      end
+    end
+    ```
+
+    The first time a `PostBox` is created, the `title` will be set to `"My new blog post-1"`. The next time
+    one is created, the `title` will be set to `"My new blog post-2"`, and so on.
+
+    > Sequences always return a `String`. For sequence type values on other types, you'll need to
+    > implement those yourself.
+
+    ## Associations with boxes
+
+    When you create a Box, you may want to have an association already set for you. In this case,
+    you'll just use that association's box to set the `foreign_key` value.
+
+    ```crystal
+    class PostBox < Avram::Box
+      def initialize
+        title sequence("post-title")
+        body "blah"
+        user_id UserBox.create.id
+      end
+    end
+    ```
+
+    > Creating a Box returns an instance of that model, which means you have access to all of the model
+    > methods.
+
     ## Saving records
 
+    As shown previously, a Box has a `create` method which saves the record to your database. A Box is
+    essentially a fancy wrapper around [SaveOperation](#{Guides::Database::ValidatingSavingDeleting.path}).
+
+    Boxes give you access to two helpful class methods `create` and `create_pair`.
+
+    ### Box.create
+
+    This will create a `Post`, and store it in your database, then return the instance of `Post`.
+    It will use the defaults you defined in your `initialize` method of `PostBox`.
+
     ```crystal
-    PostBox.create
-    PostBox.create_pair
+    post = PostBox.create
+    post.title #=> "post-title-1"
+
+    post2 = PostBox.create
+    post2.title #=> "post-title-2"
     ```
 
     ### Overriding data
 
+    When you need to override the defaults previously set, you'll pass a block to `create`.
+
+    ```crystal
+    PostBox.create do |box|
+      box.title("Draft")
+      box.body("custom body")
+    end
+    ```
+
+    Thanks to Crystal's [short one-argument syntax](https://crystal-lang.org/reference/syntax_and_semantics/blocks_and_procs.html#short-one-argument-syntax), we can shorten this with
+
     ```crystal
     PostBox.create &.title("Draft").body("custom body")
-    PostBox.create_pair &.title("Draft")
+    ```
+
+    ### Box.create_pair
+
+    It may be common for you to need to create more than 1 test object at a time. For this, you can use
+    the `create_pair` method to create 2 test objects!.
+
+    ```crystal
+    PostBox.create_pair
+    ```
+
+    > The big difference here is that `create_pair` will return `nil`, and you can't override
```

Comment:
Good point. We should make it so it returns the records and can accept a block.

https://github.com/luckyframework/avram/issues/199

This is fine for now though since we probably won't  add that issue before release. This weekend is busy for me